Gems/ROS2/Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -100,99 +102,80 @@ namespace ROS2
         // Apply transform as per origin
         PrefabMakerUtils::SetEntityTransformLocal(visual->RawPose(), entityId);
 
-        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
         auto geometry = visual->Geom();
         switch (geometry->Type())
         {
         case sdf::GeometryType::SPHERE:
             {
                 auto sphereGeometry = geometry->SphereShape();
                 AZ_Assert(sphereGeometry, "geometry is not Sphere");
-                entity->CreateComponent(LmbrCentral::EditorSphereShapeComponentTypeId);
-                entity->Activate();
-                LmbrCentral::SphereShapeComponentRequestsBus::Event(
-                    entityId, &LmbrCentral::SphereShapeComponentRequests::SetRadius, sphereGeometry->Radius());
-                LmbrCentral::EditorShapeComponentRequestsBus::Event(
-                    entityId, &LmbrCentral::EditorShapeComponentRequests::SetVisibleInGame, true);
-                entity->Deactivate();
+                const AZ::Vector3 sphereDimensions(sphereGeometry->Radius());
+
+                // The `_sphere_1x1.azmodel` is created by Asset Processor based on O3DE `PrimitiveAssets` Gem source.
+                AZ::Data::AssetId assetId;
+                const char* sphereAssetRelPath = "objects/_primitives/_sphere_1x1.azmodel"; // relative path to cache folder.
```

Comment:
I'm pretty sure this should be .fbx.azmodel if you're building against o3de/development. I'm not quite sure how this is working for you as written :( .

```suggestion
                // The `_sphere_1x1.fbx.azmodel` is created by Asset Processor based on O3DE `PrimitiveAssets` Gem source.
                AZ::Data::AssetId assetId;
                const char* sphereAssetRelPath = "objects/_primitives/_sphere_1x1.fbx.azmodel"; // relative path to cache folder.
```

Reply:
> 'm pretty sure this should be .fbx.azmodel if you're building against o3de/development. I'm not quite sure how this is working for you as written :( .

Thank you for spotting it. Indeed I tend to change O3DE less frequently to skip long compilation times. I was using one-month-old O3DE (c9db420b). When switched to a recent `stabilization` (85891e13e7) it behaves as you expected:
```bash
jhanca@jhanca-buntu ~/devroot/projects/R2PT
  % find | grep sphere_1x1
./Cache/linux/objects/_primitives/_sphere_1x1.fbx.pxmesh
./Cache/linux/objects/_primitives/_sphere_1x1_lod0_position0.fbx.azbuffer
./Cache/linux/objects/_primitives/_sphere_1x1_lod0_tangent0.fbx.azbuffer
./Cache/linux/objects/_primitives/_sphere_1x1.fbx.azmodel
./Cache/linux/objects/_primitives/_sphere_1x1_lambert3_10605252722126846652.fbx.azmaterial
./Cache/linux/objects/_primitives/_sphere_1x1_middlegrey.azmaterial
./Cache/linux/objects/_primitives/_sphere_1x1_lod0_bitangent0.fbx.azbuffer
./Cache/linux/objects/_primitives/_sphere_1x1.fbx.abdata.json
./Cache/linux/objects/_primitives/_sphere_1x1_lod0_normal0.fbx.azbuffer
./Cache/linux/objects/_primitives/_sphere_1x1_lod0_index.fbx.azbuffer
./Cache/linux/objects/_primitives/_sphere_1x1_lod0_uv0.fbx.azbuffer
./Cache/linux/objects/_primitives/_sphere_1x1_lod0.fbx.azlod
```